src/vs/workbench/api/browser/mainThreadTreeViews.ts

```diff
@@ -177,8 +177,8 @@ export class MainThreadTreeViews extends Disposable implements MainThreadTreeVie
 	private registerListeners(treeViewId: string, treeView: ITreeView): void {
 		this._register(treeView.onDidExpandItem(item => this._proxy.$setExpanded(treeViewId, item.handle, true)));
 		this._register(treeView.onDidCollapseItem(item => this._proxy.$setExpanded(treeViewId, item.handle, false)));
-		this._register(treeView.onDidChangeSelection(items => this._proxy.$setSelection(treeViewId, items.map(({ handle }) => handle))));
-		this._register(treeView.onDidChangeFocus(item => this._proxy.$setFocus(treeViewId, item.handle)));
+		this._register(treeView.onDidChangeSelection(items => this._proxy.$setSelectionAndFocus(treeViewId, this.getHandles(items), treeView.getFocus()?.handle)));
+		this._register(treeView.onDidChangeFocus(item => this._proxy.$setSelectionAndFocus(treeViewId, this.getHandles(treeView.getSelection()), item.handle)));
```

Comment:
Not as easy as I expected since we also need to internally change `ITreeView.onDidChangeSelection` and `ITreeView.onDidChangeFocus` to `ITreeView.onDidChangeSelectionOrFocus` so we can set them at once... I looked at it for a bit and it seems we need to change `Trait` class in `abstractTree.ts` such that it contains both selection and focus and triggers `onDidChange` for them both at once.

Am I missing something? If not I'd appreciate some pointers!

Reply:
I have opened https://github.com/microsoft/vscode/issues/185563. We can't do anything else to improve this for now.

Reply:
Thanks, it does seem that changing the internal tree is out of scope for this PR